staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -748,8 +748,8 @@ func (az *Cloud) SetInformers(informerFactory informers.SharedInformerFactory) {
 		UpdateFunc: func(prev, obj interface{}) {
 			prevNode := prev.(*v1.Node)
 			newNode := obj.(*v1.Node)
-			if newNode.Labels[LabelFailureDomainBetaZone] ==
-				prevNode.Labels[LabelFailureDomainBetaZone] {
+			if newNode.Labels[LabelTopologyZone] ==
```

Comment:
I am not sure exactly what it does here but you might want to check if preNode is FailureDomain and new Node is topologyZone, also keep the `newNode.Labels[LabelFailureDomainBetaZone] == prevNode.Labels[LabelFailureDomainBetaZone]` for backward compatibility maybe?